testing/helpers/index.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * A private encapsulation of the test env variable.
+ */
+let env_;
+
+/**
+ * Sets up the helper environment.
+ * @param {*} env
+ */
+export function configureHelpers(env) {
+  env_ = env;
+}
+
+/**
+ * Returns a Promise that resolves after the specified number of milliseconds.
+ * @param {number} ms
+ * @return {Promise<void>}
+ */
+export function sleep(ms) {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve();
+    }, ms);
+  });
+}
+
+/**
+ * Returns a Promise that resolves after the next browser frame has been rendered.
+ * @param {Window=} win
+ * @return {Promise<void>}
```

Comment:
In CC types this would need to be `{!Promise<void>}` since all object types are implicitly nullable. Is this not the case in TS?

Reply:
TS is the opposite, types must be explicitly declared as nullable. It might be worth continuing closure non-nullable declarations for backwards compatibility though.